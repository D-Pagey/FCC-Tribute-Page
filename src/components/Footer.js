import React, { Component } from 'react';

class Footer extends Component {
  render() {
    return (
      <div className="footer">
        <button type="button" className="more-info">
          <a href="https://en.wikipedia.org/wiki/Chris_Sharma" target="_blank" className="link">More Info</a>
        </button>
        <p class="signature copy">Written and coded by
          <a href="https://www.freecodecamp.org/d-pagey" target="_blank" className="freecodecamp copy">Pagey</a>
        </p>
      </div>
    )
  }
}

export default Footer;
