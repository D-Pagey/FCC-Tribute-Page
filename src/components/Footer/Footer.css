.footer {
  display: flex;
  flex-direction: column;
  align-items: center; }
  @media screen and (min-width: 900px) {
    .footer {
      flex-direction: row;
      justify-content: space-between;
      margin: 0 2em;
      padding-bottom: 1em; } }
  @media screen and (min-width: 2000px) {
    .footer {
      padding: 1em 0;
      margin: 0 3em; } }

.more-info {
  background-color: transparent;
  border: 1px solid #17a2b8;
  border-radius: 2px;
  color: #17a2b8;
  padding: .5em .65em;
  margin-bottom: .5em;
  width: 7em; }
  @media screen and (min-width: 2000px) {
    .more-info {
      font-size: 1em; } }
  @media screen and (min-width: 2000px) {
    .more-info {
      font-size: 1.5em; } }

.more-info .link {
  color: #17a2b8;
  font-size: 1.25em; }

.link {
  text-decoration: none;
}

@media screen and (min-width: 900px) {
  .link {
    font-size: 1.25rem; } }
@media screen and (min-width: 2000px) {
  .link {
    font-size: 1em; } }

.signature {
  align-self: center;
  margin: .5em 0 1em; }
  @media screen and (min-width: 430px) {
    .signature {
      font-size: .85rem; } }
  @media screen and (min-width: 2000px) {
    .signature {
      font-size: 1.5em; } }

.freecodecamp {
  color: #007bff;
  text-decoration: none; }
  @media screen and (min-width: 430px) {
    .freecodecamp {
      font-size: .85rem; } }
  @media screen and (min-width: 2000px) {
    .freecodecamp {
      font-size: 1em; } }
