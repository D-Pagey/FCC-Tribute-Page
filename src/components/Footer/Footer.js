import React from 'react';

import './Footer.css';

function Footer() {
  return (
    <div className="footer">
      <button type="button" className="more-info">
        <a href="https://en.wikipedia.org/wiki/Chris_Sharma"
          target="_blank" rel='noreferrer noopener' className="link">More Info</a>
      </button>
      <p className="signature">Written and coded by <a
        href="https://www.freecodecamp.org/d-pagey" target="_blank" 
        rel='noreferrer noopener' className="freecodecamp">Pagey</a>
      </p>
    </div>
  )
}

export default Footer;
