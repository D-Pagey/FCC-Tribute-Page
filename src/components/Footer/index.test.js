import React from 'react';
import { shallow } from 'enzyme';

import Footer from '.';

it('render a footer', () => {
  const wrapper = shallow(
    <Footer />,
  );
  expect(wrapper).toMatchSnapshot();
});
