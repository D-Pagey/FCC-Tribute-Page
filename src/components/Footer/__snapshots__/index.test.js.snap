// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a footer 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="more-info"
          type="button"
>
          <a
                    className="link"
                    href="https://en.wikipedia.org/wiki/Chris_Sharma"
                    rel="noreferrer noopener"
                    target="_blank"
          >
                    More Info
          </a>
</button>,
        <p
          className="signature"
>
          Written and coded by
          <a
                    className="freecodecamp"
                    href="https://www.freecodecamp.org/d-pagey"
                    rel="noreferrer noopener"
                    target="_blank"
          >
                     Pagey
          </a>
</p>,
      ],
      "className": "footer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            className="link"
            href="https://en.wikipedia.org/wiki/Chris_Sharma"
            rel="noreferrer noopener"
            target="_blank"
>
            More Info
</a>,
          "className": "more-info",
          "type": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "More Info",
            "className": "link",
            "href": "https://en.wikipedia.org/wiki/Chris_Sharma",
            "rel": "noreferrer noopener",
            "target": "_blank",
          },
          "ref": null,
          "rendered": "More Info",
          "type": "a",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Written and coded by",
            <a
              className="freecodecamp"
              href="https://www.freecodecamp.org/d-pagey"
              rel="noreferrer noopener"
              target="_blank"
>
               Pagey
</a>,
          ],
          "className": "signature",
        },
        "ref": null,
        "rendered": Array [
          "Written and coded by",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Pagey",
              "className": "freecodecamp",
              "href": "https://www.freecodecamp.org/d-pagey",
              "rel": "noreferrer noopener",
              "target": "_blank",
            },
            "ref": null,
            "rendered": " Pagey",
            "type": "a",
          },
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="more-info"
            type="button"
>
            <a
                        className="link"
                        href="https://en.wikipedia.org/wiki/Chris_Sharma"
                        rel="noreferrer noopener"
                        target="_blank"
            >
                        More Info
            </a>
</button>,
          <p
            className="signature"
>
            Written and coded by
            <a
                        className="freecodecamp"
                        href="https://www.freecodecamp.org/d-pagey"
                        rel="noreferrer noopener"
                        target="_blank"
            >
                         Pagey
            </a>
</p>,
        ],
        "className": "footer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="link"
              href="https://en.wikipedia.org/wiki/Chris_Sharma"
              rel="noreferrer noopener"
              target="_blank"
>
              More Info
</a>,
            "className": "more-info",
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "More Info",
              "className": "link",
              "href": "https://en.wikipedia.org/wiki/Chris_Sharma",
              "rel": "noreferrer noopener",
              "target": "_blank",
            },
            "ref": null,
            "rendered": "More Info",
            "type": "a",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Written and coded by",
              <a
                className="freecodecamp"
                href="https://www.freecodecamp.org/d-pagey"
                rel="noreferrer noopener"
                target="_blank"
>
                 Pagey
</a>,
            ],
            "className": "signature",
          },
          "ref": null,
          "rendered": Array [
            "Written and coded by",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Pagey",
                "className": "freecodecamp",
                "href": "https://www.freecodecamp.org/d-pagey",
                "rel": "noreferrer noopener",
                "target": "_blank",
              },
              "ref": null,
              "rendered": " Pagey",
              "type": "a",
            },
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
